### PR TITLE
using this.stringify instead of JSON.stringify

### PR DESCRIPTION
**Pull request by doron2402**
typo

### CODE REVIEW

**Comment by mcollina** [comment]
Can you add a unit test for this?
What's the problem this is causing you?

**Comment by jsumners** [comment]
I must admit, I'm confused by this as well. You are claiming this is a typo, but Pino's prototype chain is `EventEmitter => Function => Object`. Thus, `.stringify` isn't even a member of the Pino object.

**Comment by doron2402** [comment]
@mcollina 
- I can add a unit test.
- no problem, just don't want to use `JSON.stringify`

@jsumners 
 - just trying to keep it more consistency,  it uses ` this.stringify` in line 138 and 143.

**Comment by jsumners** [comment]
Okay, I forgot about https://github.com/pinojs/pino/blob/e6fbe72f14b9683abcbff95039c4ceb4db89021d/pino.js#L270

I agree this PR should proceed.

**Comment by doron2402** [comment]
@jsumners 
```
Object.defineProperty(pinoPrototype, 'asJson', {
  enumerable: true,
  value: asJson
})
```

**Comment by jsumners** [comment]
@doron2402 what is your point?

**Comment by doron2402** [comment]
to your comment: Thus, .stringify isn't even a member of the Pino object.

**Comment by jsumners** [comment]
@doron2402 the code you quoted doesn't actually impart the `.stringify` method. It's the line I linked in my previous response that does. I had merely forgot we have that option.

**Comment by doron2402** [comment]
should the test live under `test/escaping.test.js` ?

**Comment by mcollina** [comment]
@doron2402 would you mind adding that unit test? We need it to merge this.

**Comment by jsumners** [comment]
@doron2402 since @mcollina didn't answer your question, put the test wherever you think it makes sense.

**Comment by mcollina** [comment]
I think `test/basic.test.js` is ok.

**Comment by doron2402** [comment]
cool will do, I'll add it 
thanks @jsumners @mcollina

File: pino.js [hunk 1]
```diff
@@ -267,7 +266,7 @@ function pino (opts, stream) {
   }
 
   // internal options
-  iopts.stringify = iopts.safe ? stringifySafe : JSON.stringify
+  iopts.stringify = iopts.safe ? require('fast-safe-stringify') : JSON.stringify
```
Comment (mcollina) [inline pino.js]: can you move this back to the top?

File: test/basic.test.js [hunk 2]
```diff
@@ -442,3 +443,25 @@ test('children with same names render in correct order', function (t) {
 
   root.child({a: 1}).child({a: 2}).info({a: 3})
 })
+
+//  https://github.com/pinojs/pino/pull/251 - use this.stringify
```
Comment (mcollina) [inline test/basic.test.js]: there is a double space here

Comment (doron2402) [inline test/basic.test.js]: 💯

File: pino.js (hunk 1 above)
Comment (doron2402) [inline pino.js]: 👍

**Comment by doron2402** [comment]
done!

Thanks @mcollina

**Pull request merged**